mapper/transaction.go

```diff
@@ -88,93 +90,114 @@ func CrossChainTransactions(
 		return transactions, nil
 	}
 
-	tx := &evm.Tx{}
-	if _, err := codecManager.Unmarshal(extra, tx); err != nil {
-		return transactions, err
+	// Initialize Atomic Transactions
+	atomicTxs := make([]*evm.Tx, 0)
+	if block.Time() < AP5Activation {
+		// Prior to Apricot Phase 5, there was only one atomic transaction per
+		// block.
+		tx := new(evm.Tx)
+		if _, err := codecManager.Unmarshal(extra, tx); err != nil {
+			return nil, err
+		}
+		atomicTxs = append(atomicTxs, tx)
+	} else {
+		if _, err := codecManager.Unmarshal(extra, &atomicTxs); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal atomic tx (AP5) due to %w", err)
+		}
```

Comment:
nit: rather than initializing an empty array (that is never used) I feel like it would be cleaner to do:
```
var atomicTxs []*evm.Tx
if block.Time() < AP5Activation {
	// Prior to Apricot Phase 5, there was only one atomic transaction per
	// block.
	tx := new(evm.Tx)
	if _, err := codecManager.Unmarshal(extra, tx); err != nil {
		return nil, err
	}
	atomicTxs = []*evm.Tx{tx}
} else {
	if _, err := codecManager.Unmarshal(extra, &atomicTxs); err != nil {
		return nil, fmt.Errorf("failed to unmarshal atomic tx (AP5) due to %w", err)
	}
}
```